frontend/src/components/Recipes/Sidebar/SearchSection/SearchSection.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+import { SearchTerm, SelectCuisine, SelectPreference } from '.'
+import { useSearchParams } from 'react-router'
+
+import { getNonEmptySearchParams } from '../../../../utils/searchParams'
+
+export default function SearchSections() {
+    const [searchParams, setSearchParams] = useSearchParams()
+
+    const [term, setTerm] = useState(searchParams.get('q') || '')
+    const [cuisineId, setCuisineId] = useState(
+        searchParams.get('cuisineId') || ''
+    )
+    const [dietId, setDietId] = useState(searchParams.get('dietId') || '')
+
+    const handleSearch = () => {
+        setSearchParams(
+            getNonEmptySearchParams({
+                q: term,
+                cuisineId,
+                dietId,
+                difficultyId: searchParams.get('difficultyId') || '',
+            })
+        )
+    }
+
+    useEffect(() => {
+        setTerm(searchParams.get('q') || '')
+        setCuisineId(searchParams.get('cuisineId') || '')
+        setDietId(searchParams.get('dietId') || '')
+    }, [searchParams])
```

Comment:
This is the useEffect needed to clear each state when navigating back to the /recipes from the header.

By logging all the possible states it seems like the params are null, while the internal component state still has values populated.

This seems to happen because when clicking the link from the AppHeader, no re-render is triggered inside the Sidebar components.
I need the useEffect with searchParams inside the deps array to trigger a repopulation of the setState variables.